### DISCUSSION TITLE
Google's x/tools (of gopls fame) added a new deadcode tool recently

### DISCUSSION THREAD

**Discussion by tombh** [golangci/golangci-lint#3977] [General] [answered]
From [their README](https://cs.opensource.google/go/x/tools/+/master:internal/cmd/deadcode/doc.go):
> The deadcode command loads a Go program from source then uses Rapid
Type Analysis (RTA) to build a call graph of all the functions
reachable from the program's main function. Any functions that are not
reachable are reported as dead code, grouped by package.

_tl;dr_ this is like `unused` but for an entire application.

Once it stabilises might it be a good candidate for `golangci-lint`?

Sorry if this has been addressed already, I did have a search but couldn't see anything.

**Comment by ldez** [comment, marked as answer]
Hello,

`deadcode` is a pure command line (package `main` only => cannot be used as a lib) and not based on `go/analysis` API.

golangci-lint is based on the `go/analysis` API to optimize the runs of the different linters.

The idea is interesting but it's not a good candidate for golangci-lint.
